### ISSUE TITLE
linter: `unicorn/prefer-array-flat-map` misses `.filter().flatMap()`

### ISSUE THREAD

**Issue by lukpsaxo** [oxc-project/oxc#26594]
### What version of Oxlint are you using?

1.82.0

### What command did you run?

oxlint

### What does your `.oxlintrc.json` (or `oxlint.config.ts`) config file look like?

```json
{
    "plugins": ["unicorn"],
    "rules": {
        "unicorn/prefer-array-flat-map": "error"
    }
}
```

### What happened?

```js
const result = values
    .filter((value) => value > 0)
    .flatMap((value) => [value, value]);
```

## Actual behavior

Oxlint reports no `unicorn/prefer-array-flat-map` diagnostic.

## Expected behavior

Oxlint should report:

> Prefer a single `.flatMap(…)` over `.filter(…).flatMap(…)`.

It should offer the same suggestion as eslint-plugin-unicorn:

```js
const result = values.flatMap((value) => (value > 0 ? [value, value] : []));
```

eslint-plugin-unicorn's rule covers both `.map().flat()` and `.filter().flatMap()` chains. The
second form is reported only when both callbacks are simple, untyped, single-parameter arrow
functions using the same parameter and the `flatMap` callback returns an array with multiple items.
Those restrictions make the suggested merge safe for the supported pattern.

Upstream rule documentation:

https://github.com/sindresorhus/eslint-plugin-unicorn/blob/v74.0.0/docs/rules/prefer-array-flat-map.md
